pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -1047,58 +988,24 @@ async def parse_packages_components(self, comps_result):
 
     async def parse_packages(self, primary_xml, filelists_xml, other_xml, file_extension="gz"):
         """Parse packages from the remote repository."""
-        packages = await RpmFirstStage.parse_repodata(
+        parser = MetadataParser.from_metadata_files(
             primary_xml.path, filelists_xml.path, other_xml.path
         )
 
-        # skip SRPM if defined
-        if "srpm" in self.skip_types and not self.mirror:
-            packages = collections.OrderedDict(
-                (pkgId, pkg) for pkgId, pkg in packages.items() if pkg.arch != "src"
-            )
```

Comment:
This was a bug: We were skipping the SRPMs on one side but still running into them on the iterative side, which means they'd get coded as "not seen before" and trigger the assert.